jep/0000/README.adoc

```diff
@@ -0,0 +1,227 @@
+
+// JENKINS-52342 for ^ which is the design, but not done fully yet.
+
+As a general rule for Essentials, we are using the `configuration-as-code` plugin to do the necessary configuration.
+
+*No credentials* are specified.
+The permissions are retrieved dynamically through the instance profile used when creating the EC2 instance for the Essentials master.
+
+=== Instance Profile and IAM role
+
+A dedicated instance profile is created by the _CloudFormation_ template to allow creating EC2 instances for agents, and access a specific S3 bucket to store artifacts and Pipeline stashes.
+
+==== EC2 policy
+
+[source,json]
+"Statement": [{
+    "Sid": "ControlEC2JenkinsAgents",
+    "Effect": "Allow",
+    "Action": [
+        "ec2:DescribeInstances",
+        "ec2:TerminateInstances",
+        "ec2:RequestSpotInstances",
+        "ec2:DeleteTags",
+        "ec2:CreateTags",
+        "ec2:DescribeRegions",
+        "ec2:RunInstances",
+        "ec2:DescribeSpotInstanceRequests",
+        "ec2:StopInstances",
+        "ec2:DescribeSecurityGroups",
+        "ec2:GetConsoleOutput",
+        "ec2:DescribeImages",
+        "ec2:CancelSpotInstanceRequests",
+        "ec2:StartInstances",
+        "ec2:DescribeAvailabilityZones",
+        "ec2:DescribeSubnets",
+        "ec2:DescribeKeyPairs"
+    ],
+    "Resource": "*"
+}]
+
+==== S3 bucket policy
+
+FIXME: refine through JENKINS-52342.
+
+[source,json]
+"Statement": [{
+        "Sid": "TodoRefineSecurityALot",
+        "Effect": "Allow",
+        "Action": [
+            "s3:PutObject",
+            "s3:GetObject",
+            "s3:ListBucket",
+            "s3:DeleteObject"
+        ],
+        "Resource": "*"
+    }
+]
+
+=== Security groups
+
+==== Master
+A dedicated group is created to allow access to the following ports:
+
+22:: restricted to an optional parameter.
+Users would typically use `curl ident.me` to avoid making this port accessible to the world.
+8080:: not restricted.
+
+==== Agents
+
+A dedicated group is created to allow only access to port `22`.
+
+== Motivation
+
+Nothing was existing to provide an autoconfigured setup of Jenkins in a specific Cloud environment.
+
+== Reasoning
+
+=== Instance Profile
+
+Early during the prototype, a dedicated service user was used.
+This was replaced by instance profile later for various reasons.
+
+This is because it is the recommended path for AWS, but also because doing so makes the configuration leaner:
+both the _EC2_ and _Artifact Manager on S3_ plugins will autodetect their permissions if no explicit credentials are specified.
+
+== Backwards Compatibility
+
+There are no backwards compatibility concerns related to this proposal.
+
+== Security
+
+=== Allowed certificates
+
+link:https://github.com/jenkinsci/jep/tree/master/jep/307#security[JEP 307] aggressively restricts the list of certificates that will be allowed to be used from inside Essentials container.
```

Comment:
Seems like a poor design choice in JEP-307. It should not touch the JRE-wide certificate list. Rather, the one piece of code which actually retrieves an update manifest should specify the root certificates it trusts.

Anyway I thought that this transmission was to the Evergreen client rather than to Jenkins itself, so why would this have any effect on Jenkins plugins? This is not making sense to me.

Reply:
We do not touch the JRE certificates, we modify the whole Docker image certificates. So it affects the whole system.

See https://github.com/jenkins-infra/evergreen/blob/a949d54575517f1e29885043bcff21617a70c6c9/distribution/scripts/generate-ca-certificates

Reply:
> Seems like a poor design choice in JEP-307

Care to develop that statement? I think this is actually a good thing to aggressively reduce the set of targets we allow things to access from the master container. This would make it indeed impractical to run workload from the master and do anything from there (even https://github.com would be considered insecure), but that is actually almost a feature as we do **not** want to run things on the master.